crates/pixi_glob/src/glob_set.rs

```diff
@@ -72,25 +75,55 @@ impl<'t> GlobSet<'t> {
             .include
             .iter()
             .flat_map(move |glob| {
-                glob.walk(root_dir.clone())
+                let (effective_walk_root, glob) = if glob.has_semantic_literals() {
```

Comment:
this is basically new behaviour.

Reply:
is that check really necessary?
Can't you always use `partition`?

Reply:
Yes - let's take an example that we have `pixi.toml` glob

Calling partition on it, we will receive
`pixi.toml` prefix and empty glob, which when `walking`  will not match.